source/common/http/utility.cc

```diff
@@ -32,6 +32,28 @@
 #include "nghttp2/nghttp2.h"
 
 namespace Envoy {
+namespace Http {
+namespace Utility {
+Http::Status exceptionToStatus(std::function<Http::Status(Buffer::Instance&)> dispatch,
+                               Buffer::Instance& data) {
+  Http::Status status;
+  try {
+    status = dispatch(data);
+    // TODO(#10878): Remove this when exception removal is complete. It is currently in migration,
+    // so dispatch may either return an error status or throw an exception. Soon we won't need to
+    // catch these exceptions, as all codec errors will be migrated to using error statuses that are
+    // returned from dispatch.
+  } catch (FrameFloodException& e) {
+    status = bufferFloodError(e.what());
+  } catch (CodecProtocolException& e) {
+    status = codecProtocolError(e.what());
+  } catch (PrematureResponseException& e) {
+    status = prematureResponseError(e.what(), e.responseCode());
```

Comment:
should we also catch EnvoyException to make sure there are no others that leak through?

or maybe even std::exception?

Reply:
I would rather not catch any more exceptions than we do today and otherwise let the process crash? I'm not sure why we would expand the catch targets as part of this cleanup?

Reply:
sg. if we don't have a higher level catch then expanding the scope here isn't needed.